### PR TITLE
Gun fire delay, grille, and wall mount adjustments.

### PR DESCRIPTION
**Pull request by PetMudstone**
### Purpose
- Replaces the FireRate variable in Gun.cs with FireDelay, which is the time in seconds that a gun can fire again as opposed to FireRate being how many times a gun can fire in a second. Done as FireDelay is more similar to how /tg/ (and CM) does it.
  - Tweaks for fire delays across the board.
- Makes broken grilles less durable. Stops bullets from colliding with them.
- _Hopefully_ makes wall-mounted objects no longer block projectiles
- Makes air meters no longer block movement.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Weapons/Gun.cs [hunk 1]
```diff
@@ -130,10 +130,10 @@ protected StandardProgressActionConfig ProgressConfig
 		public AddressableAudioSource DryFireSound;
 
 		/// <summary>
-		/// The amount of times per second this weapon can fire
+		/// The time in seconds before this weapon can fire again.
 		/// </summary>
-		[Tooltip("The amount of times per second this weapon can fire")]
-		public double FireRate;
+		[Tooltip("The time in seconds before this weapon can fire again.")]
+		public double FireDelay = 0;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Weapons/Gun.cs]: is zero a good default delay?

Comment (PetMudstone) [reply UnityProject/Assets/Scripts/Items/Weapons/Gun.cs]: Honestly, no. Will change to 0.5 right now (which is similar to how fast guns with 0 delay fire in /tg/ due to lag and stuff, actually).

File: UnityProject/Assets/Scripts/Items/Weapons/Gun.cs [hunk 2]
```diff
@@ -842,7 +842,7 @@ private void DequeueAndProcessServerShot()
 			if (shooter == PlayerManager.LocalPlayer)
 			{
 				//this is our gun so we need to update our predictions
-				FireCountDown += 1.0 / FireRate;
+				FireCountDown = FireDelay;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Weapons/Gun.cs]: I remember something about Autumn saying this should keep the +=

Comment (PetMudstone) [reply UnityProject/Assets/Scripts/Items/Weapons/Gun.cs]: Plus sign added (properly this time).

**Comment by Bod9001** [comment]
can't work out why tests are failing Shall I merge?

**Comment by ThatDan123** [comment]
maybe open and run it in editor just to make sure there isnt a compilation error

**Comment by corp-0** [comment]
if it has been reran and still failed, then most likely it has compiling errors.

**Pull request merged**